lib/AST/ASTPrinter.cpp

```diff
@@ -4414,12 +4414,18 @@ swift::getInheritedForPrinting(const Decl *decl,
     inherited = ed->getInherited();
   }
 
-  // Collect explicit inheritted types.
+  // Collect explicit inherited types.
   for (auto TL: inherited) {
-    if (auto Ty = TL.getType()) {
-      if (auto NTD = Ty->getAnyNominal())
-        if (!shouldPrint(NTD))
-          continue;
+    if (auto ty = TL.getType()) {
+      bool foundUnprintable = ty.findIf([shouldPrint](Type subTy) {
```

Comment:
Can you factor this out? Especially if it's gonna be reused in #20146.

Reply:
It's not exactly the same query, though: the `shouldPrint` here is an argument to the function the way ASTPrinter is currently written. I suppose Type itself could have an overload of `findIf` that passes GenericTypeDecls.